simphony/testing/abc_check_engine.py

```diff
@@ -177,15 +177,17 @@ def test_delete_dataset(self):
             name = "test_" + str(i)
             engine.add_dataset(self.create_dataset(name=name))
 
+        datasets_and_names = [(ds, ds.name) for ds in engine.iter_datasets()]
+
         # delete each of the datasets
-        for ds in engine.iter_datasets():
-            engine.remove_dataset(ds.name)
+        for (dataset, name) in datasets_and_names:
+            engine.remove_dataset(name)
             # test that we can't get deleted datasets
             with self.assertRaises(ValueError):
-                engine.get_dataset(ds.name)
+                engine.get_dataset(name)
             # test that we can't use the deleted datasets
             with self.assertRaises(Exception):
-                self.compare_dataset(ds, ds)
+                self.compare_dataset(dataset, dataset)
```

Comment:
It works for `simphony-lammps-md` (https://github.com/simphony/simphony-lammps-md/blob/master/simlammps/tests/test_lammps_wrapper.py).

I take a look at simphony-jyulb in the next few days.

-------------

>  I tested it with Jyulb wrapper and this particular line causes a Segmentation Fault without any other messages. When I check the type of dataset before the compare_datasets call, it reports a ProxyLattice object. 

Can you recreate this (i.e. have some code that causes the segmentation fault)?   Why don't you debug it too see when the segmentation fault occurs. 

Lets discuss this on a different issue in simphony-jyulb.